core/src/main/java/com/sequenceiq/cloudbreak/service/parcel/ParcelFilterService.java

```diff
@@ -39,26 +39,30 @@
     @Inject
     private RestClientFactory restClientFactory;
 
-    public Set<ClouderaManagerProduct> filterParcelsByBlueprint(Set<ClouderaManagerProduct> parcels, Blueprint blueprint) {
+    @Inject
+    private ParcelInfoProvider parcelInfoProvider;
+
+    public Set<ClouderaManagerProduct> filterParcelsByBlueprint(Long stackId, Set<ClouderaManagerProduct> parcels, Blueprint blueprint) {
+        LOGGER.debug("Filtering the following parcels based on the blueprint {}", parcels);
         Set<String> serviceNamesInBlueprint = getAllServiceNameInBlueprint(blueprint);
         LOGGER.debug("The following services are found in the blueprint: {}", serviceNamesInBlueprint);
         Set<ClouderaManagerProduct> ret = new HashSet<>();
         if (serviceNamesInBlueprint.contains(null)) {
             LOGGER.debug("We can not identify one of the service from the blueprint so to stay on the safe side we will add every parcel");
             return parcels;
         }
```

Comment:
I know, this is not the subject of this PR, but could you pls cover this part with unit test, too?